components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -36,14 +36,14 @@ internal const val MINUTE_IN_MS = 60 * 1000
  * before a refresh is attempted. Defaults to -1, meaning no cache is being used by default.
  */
 class ContileTopSitesProvider(
-    private val context: Context,
+    context: Context,
     private val client: Client,
     private val endPointURL: String = CONTILE_ENDPOINT_URL,
     private val maxCacheAgeInMinutes: Long = -1
 ) : TopSitesProvider {
 
+    private val applicationContext = context.applicationContext
```

Comment:
This all looks good to me now with the context things figure out! Did you want to address / handle https://github.com/mozilla-mobile/fenix/pull/23781#discussion_r808334535 here as well or in a separate PR. I think the Fenix integration is dependent on it.

Reply:
I am gonna do that separately.